drivers/sensor/esp32_temp/esp32_temp.c

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2022 Espressif Systems (Shanghai) Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT espressif_esp32_temp
+
+#include <driver/temp_sensor.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/sensor.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(esp32_temp, CONFIG_SENSOR_LOG_LEVEL);
+
+#if CONFIG_IDF_TARGET_ESP32
+#error "Temperature sensor not supported on ESP32"
```

Comment:
Wouldn't it be better to add a dependency to the Kconfig?

<pre>
depends on !CONFIG_IDF_TARGET_ESP32
</pre>

Reply:
I agree, this can be added into Kconfig file.

Reply:
thanks @bbilas. I try to make this restrictions visible to the end users so it doesn't come in a cryptic way, therefore I decide to have this ifdef. However I updated the Kconfig as well together with sample code.